hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/RSRpcServices.java

```diff
@@ -3498,6 +3501,18 @@ public ScanResponse scan(final RpcController controller, final ScanRequest reque
       throw new ServiceException(e);
     }
     HRegion region = rsh.r;
+    // If the start row is the actual start Row of the region and end row is actual end row of region
+    // or
+    // If the stop is not specified or the stop Row is greater than the end key of the current region
+    if ((Bytes.equals(rsh.s.getStartRow(), region.getRegionInfo().getStartKey())
+      && Bytes.equals(rsh.s.getStopRow(), region.getRegionInfo().getEndKey()))
+      || Bytes.equals(rsh.s.getStopRow(), HConstants.EMPTY_END_ROW)
+      || (Bytes.compareTo(region.getRegionInfo().getEndKey(), rsh.s.getStopRow()) <= 0
+      && !Bytes.equals(region.getRegionInfo().getEndKey(), HConstants.EMPTY_END_ROW))) {
+      if(!region.getTableDescriptor().isMetaTable()) {
```

Comment:
Why exclude meta table scans? Do this check first before doing all the above byte array compares?

Reply:
As @ramkrish86 mentioned above this is to get insights into possible inadvertent user errors, hence skipping meta. I will sync with Ram to figure out if we should exclude other system tables as well and then move the check accordingly.

Reply:
Ya better to avoid all system table checks

Reply:
Is this check going to happen for every scan#next calls? We should do this only once for 1st req which create the scanner on region.  Also will this code get executed for get calls also?

Reply:
Currently the number of scan requests that we get is also based on the scan() call that happens at the RsRpcServices. Also if a scan query is spanning across two regions - we wil have 2 RPC requests coming in and among that 1 will be a full region scan. That should still make sense right? Among totalscan Rpc requests how much are full region scans? 
This is only for scans and not for gets. Should we include gets also - that is generally with fixed start and stop row correct?

Reply:
region.getRegionInfo().getTable().isSystemTable(). Check for this first before you check for fullRegionScan. That will ensure we do it for the user regions only.

Reply:
`Is this check going to happen for every scan#next calls? We should do this only once for 1st req which create the scanner on region.`
I thought about this again. We can account it once only once but if say a scanner calls next() 10 times and every time RsRpcServices#scan() gets called we account the rpc scan request as 10. Now if we account for the new metric only once  then it means 1 out of every 10 scan request is full region scan. Then we are not having a ratio which we want. Instead if we had accounted all 10 as full regoin scan then we can atleast have the correct ratio. Correct me if am wrong.

Reply:
This is now moved to HRegion layer. We check it once when we instantiate the scanner.